src/api/app/mixins/webui/has_flags.rb

```diff
@@ -92,6 +92,44 @@ def get_build_flags
     return build_flags
   end
 
+  def existing_flag(options = {})
+    main_object.flags.where(
+      repo: options[:repo], architecture: options[:architecture]
+    ).first
+  end
+
+  def temporary_flag(attributes = {})
+    attributes.update(flag: 'build') # make it configurable
+    attributes[:status] = Flag.default_state('build') unless attributes[:status]
+
+    main_object.flags.new(attributes)
+  end
+
+  def fetch_flag(options = {})
+    existing_flag(options) || temporary_flag(options)
+  end
+
+  def get_build_flags_alt
+    flags = { "all" => [] }
+    flags['all'] << fetch_flag
+
+    status = flags['all'].first.status
+
+    main_object.architectures.each do |arch|
+      flags['all'] << fetch_flag(architecture: arch, status: status)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.